plugins/inputs/mesos/mesos.go

```diff
@@ -510,6 +518,14 @@ func (m *Mesos) gatherMainMetrics(a string, defaultPort string, role Role, acc t
 		return err
 	}
 
+	if role == MASTER {
+		if jf.Fields["master/cpus_total"] != 0.0 {
+			tags["state"] = "leader"
+		} else {
+			tags["state"] = "follower"
+		}
+	}
+
 	acc.AddFields("mesos", jf.Fields, tags)
```

Comment:
I've just noticed that slave metrics have timestamp which we can use to pass as fourth parameter to `AddFields`